localstack/services/secretsmanager/provider.py

```diff
@@ -0,0 +1,261 @@
+import json
+import logging
+
+from localstack.aws.api import HttpRequest, RequestContext
+from localstack.aws.api.secretsmanager import (
+    AddReplicaRegionListType,
+    BooleanType,
+    CancelRotateSecretResponse,
+    ClientRequestTokenType,
+    CreateSecretResponse,
+    DeleteResourcePolicyResponse,
+    DeleteSecretResponse,
+    DescribeSecretResponse,
+    DescriptionType,
+    GetResourcePolicyResponse,
+    GetSecretValueResponse,
+    KmsKeyIdType,
+    ListSecretVersionIdsResponse,
+    MaxResultsType,
+    NameType,
+    NextTokenType,
+    NonEmptyResourcePolicyType,
+    PutResourcePolicyResponse,
+    PutSecretValueResponse,
+    RecoveryWindowInDaysType,
+    RemoveRegionsFromReplicationResponse,
+    RemoveReplicaRegionListType,
+    ReplicateSecretToRegionsResponse,
+    RestoreSecretResponse,
+    RotateSecretResponse,
+    RotationLambdaARNType,
+    RotationRulesType,
+    SecretBinaryType,
+    SecretIdType,
+    SecretsmanagerApi,
+    SecretStringType,
+    SecretVersionIdType,
+    SecretVersionStagesType,
+    SecretVersionStageType,
+    StopReplicationToReplicaResponse,
+    TagKeyListType,
+    TagListType,
+    UpdateSecretResponse,
+    UpdateSecretVersionStageResponse,
+    ValidateResourcePolicyResponse,
+)
+from localstack.services.moto import call_moto
+from localstack.services.secretsmanager.secretsmanager_patches import apply_patches
+from localstack.utils.aws import aws_stack
+from localstack.utils.common import to_str
+
+LOG = logging.getLogger(__name__)
+
+
+class SecretsmanagerProvider(SecretsmanagerApi):
+    def __init__(self):
+        apply_patches()  # Adds missing moto attributes.
+
+    @staticmethod
+    def __transform_context_secret_id(context: RequestContext) -> RequestContext:
```

Comment:
nit: We usually avoid using _dunderscore_ (double underscore) method names, as they have specific semantics, and may have unintended side effects. Unless there is a strong reason for it here, I'd rename this to `_transform_context_secret_id(..)`.

---
Also, I think we should avoid fiddling with the raw request in the ASF provider implementations as much as possible. This PR will introduce a new utility `call_moto_with_request(..)` that we can use to send updated requests to moto: https://github.com/localstack/localstack/pull/5519 . This utility, combined with the parsed `secret_id` that is already available in the relevant API methods below, can help us to create an updated request (with the modified `secret_id`) that we can then send to moto (without having to modify the `context` or `request` directly). 👍